cardano-cli/src/Cardano/CLI/Types/Output.hs

```diff
@@ -363,14 +363,65 @@ renderScriptCosts
   -- ^ Post execution cost calculation mapping of script witness
   -- index to execution units.
   -> Either PlutusScriptCostError [ScriptCostOutput]
-renderScriptCosts (UTxO utxo) eUnitPrices scriptMapping executionCostMapping =
+renderScriptCosts (UTxO utxo) eUnitPrices scriptMapping =
+  renderScriptCostsWithScriptHashesFunc eUnitPrices getHashForScriptWitnessIndex
+ where
+  getHashForScriptWitnessIndex
+    :: ScriptWitnessIndex -> Either PlutusScriptCostError (Maybe ScriptHash)
+  getHashForScriptWitnessIndex sWitInd = case lookup sWitInd scriptMapping of
+    Nothing -> Left (PlutusScriptCostErrPlutusScriptNotFound sWitInd)
+    Just script -> anyScriptWitnessToHash script
+
+  anyScriptWitnessToHash :: AnyScriptWitness era -> Either PlutusScriptCostError (Maybe ScriptHash)
+  anyScriptWitnessToHash = \case
+    AnyScriptWitness SimpleScriptWitness{} -> Right Nothing
+    AnyScriptWitness (PlutusScriptWitness _ pVer (PScript pScript) _ _ _) ->
+      Right $ Just $ hashScript $ PlutusScript pVer pScript
+    AnyScriptWitness (PlutusScriptWitness _ _ (PReferenceScript refTxIn) _ _ _) ->
```

Comment:
You can use https://cardano-api.cardano.intersectmbo.org/cardano-api/internal/Cardano-Api-Script.html#v:getScriptWitnessReferenceInputOrScript instead of pattern matching yourself. 

Also, why 
```haskell
AnyScriptWitness SimpleScriptWitness{} -> Right Nothing
```
?
There's a reference input in `SimpleScriptWitness _ (SReferenceScript txIn)` - is it omitted accidentally, or on purpose?

Reply:
I understand that `SimpleScripts` do not have a cost, so we don't need to calculate it. But I am basically copying from the original `renderScriptCosts`

Reply:
Also, precisely for that reason, in case of reference, [getScriptWitnessReferenceInputOrScript](https://cardano-api.cardano.intersectmbo.org/cardano-api/internal/src/Cardano.Api.Script.html#getScriptWitnessReferenceInputOrScript) loses information about whether the referenced script is simple or plutus, so we would have to fail unnecessarily if a simple script used as a reference is not provided

Reply:
But we will be deleting this bit of code when we remove the deprecated parameter anyway

Reply:
Actually suggestion by @Jimbo4350 already gets rid of this: https://github.com/IntersectMBO/cardano-cli/pull/1031/commits/3604f0c7e1872fe8c78850fafb0eac69e5eb4eba